File: gym-backend/src/main/java/com/bequre/pojo/Course.java
package com.bequre.pojo;


import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import java.time.LocalDate;


@Data
@Schema(description = "课程实体")
@TableName("courses")
public class Course {

    @NotNull
    @Schema(name = "id", description = "ID")
    private Integer id;

    @NotNull
    @Schema(name = "name", description = "课程名称")
    private String name;

    @NotNull
    @Schema(name = "description", description = "课程描述")
    private String description;

    @NotNull
    @Schema(name = "coachId", description = "私教id")
    private Integer coachId;

    @NotNull
    @Schema(name = "startDate", description = "开始日期")
    private LocalDate startDate;

    @NotNull
    @Schema(name = "endDate", description = "结束日期")
    private LocalDate endDate;

    @NotNull
    @Schema(name = "daysOfWeek", description = "每周周几上课")
    private String daysOfWeek;

    @NotNull
    @Schema(name = "price", description = "价格")
    private Float price;



}

File: gym-front/src/types/course.ts
/**
 * Course，课程实体
 */
export interface Course {

    /**
     * ID
     */
    id?: string;

    /**
     * 课程名称
     */
    name: string;

    /**
     * 课程描述
     */
    description: string;

    /**
     * 私教id
     */
    coachId: string;

    /**
     * 开始日期
     */
    startDate: string;

    /**
     * 结束日期
     */
    endDate: string;

    /**
     * 每周周几上课
     */
    daysOfWeek: string;

    /**
     * 价格
     */
    price: string;

}

/**
 * 课程分页条件查询实体
 */
export interface CourseQuery {
    /**
     * 课程名称
     */
    name?: string | null;

    /**
     * 页码
     */
    pageNo: number;

    /**
     * 每页条数
     */
    pageSize: number;

}

/**
 * 带有私教名称的课程
 */
export interface CourseWithCoachName extends Course {
    coachName?: string; // 可选属性，因为我们可能找不到所有课程的教练
}